mewarpx/mewarpx/util.py

```diff
@@ -156,16 +158,119 @@ def mkdir_p(path):
+
+    if (emission_type != 'thermionic') and not np.isclose(transverse_fac, 1.0):
+        return ValueError('transverse_fac is a support argument only for '
+                            + 'thermionic emissiion models!')
+
+    if rseed is not None:
+        nprstate = np.random.get_state()
+        np.random.seed(rseed)
+    sigma = np.sqrt(constants.kb_J * T / m)
+
+    if transverse_fac < 0.:
+        print("WARNING: transverse_fac is out of bounds")
+        print("Constraining to minimum value of 0.")
+        beta = 0.
+    elif transverse_fac > 2.:
+        print("WARNING: transverse_fac is out of bounds")
+        print("Constraining to maximum value of 2.")
+        beta = np.sqrt(2.)
+    else:
+        beta = np.sqrt(transverse_fac)
+
+    alpha = np.sqrt(2. - beta**2.)
+
+    # vx and vy follow Maxwellian distributions
+    vx = sigma * np.random.randn(num_samples) * beta
+    vy = sigma * np.random.randn(num_samples) * beta
+
+    if emission_type == 'random':
+        vz = sigma * np.random.randn(num_samples) * beta
+    elif emission_type == 'thermionic':
+        # vz is truncated with k_B*T/m << Phi assumed. See
+        # "Emission Distribution from a Thermionic Cathode" document
+        # on Bloomfire.
+        P = np.random.rand(num_samples)
+        vz = np.sqrt(-2 * sigma**2 * np.log(1 - P)) * alpha
+    elif emission_type == 'half_maxwellian':
+        vz = np.abs(sigma * np.random.randn(num_samples) * beta)
+    else:
+        raise ValueError('Unsupported emission type "%s".' % emission_type)
+
+    if rseed is not None:
+        np.random.set_state(nprstate)
+    return vx, vy, vz
+
+
+def get_positions(num_samples, xmin, xmax, ymin=0, ymax=0, z=0,
+                rseed=None):
+
+    """Provide random samples of [x, y, z] for electrons in simulation.
+    In x and y, positions are uniformly distributed. In z, positions are
+    placed at the emitter.
+    Arguments:
+        num_samples (int): Number of particles to generate positions for
+        xmin (float): Min position in x (meters)
+        xmax (float): Max position in x (meters)
+        ymin (float): Min position in y (meters)
+        ymax (float): Max position in y (meters)
+        z (float): Position of the emitter on the z-axis (meters)
+        rseed (positive int): If specified, seed the random number generator.
+            Used for testing. The random number generator is set back at the
+            end of the function.
+    Returns:
+        positions (np.ndarray): Array of shape (num_samples, 3) with positions.
     """
+    if rseed is not None:
+        nprstate = np.random.get_state()
+        np.random.seed(rseed)
+
+    # Random x and y positions
+    x = xmin + (xmax - xmin) * np.random.rand(num_samples)
+    y = ymin + (ymax - ymin) * np.random.rand(num_samples)
+    z = np.ones_like(x) * z
+
+    if rseed is not None:
+        np.random.set_state(nprstate)
+
+    return x, y, z
     return (p * constants.torr_cgs) / (constants.kb_cgs * T)
```

Comment:
I'm guessing this is due to some merging issue?
```suggestion
```

Reply:
Oops, yeah good catch, this was from merging.